### PR TITLE
Update behavior of FromLegacy to make more flexible with materials

### PR DESCRIPTION
**Pull request by errissa**
## Type

-   [ ] Bug fix (non-breaking change which fixes an issue): Fixes #
-   [x] New feature (non-breaking change which adds functionality). Resolves #
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) Resolves #

## Motivation and Context

Current behavior of `FromLegacy` on a legacy `TriangleMesh` with more than one material is to warn the user that Tensor-based meshes don't support multiple materials and not convert _any_ materials. Annoyingly, sometimes our importers (mostly ASSIMP) import or create dummy materials so converting those to Tensor-based `TriangleMesh` result in meshes with no materials. With this PR those cases will convert to a Tensor-based `TriangleMesh` with the first material converted as well. In most cases this should produce a mesh with a usable material. In any case, a message is output to the user warning that only the first material was converted.

## Checklist:

-   [X] I have run `python util/check_style.py --apply` to apply Open3D **code style**
    to my code.
-   [ ] This PR changes Open3D behavior or adds new functionality.
    -   [ ] Both C++ (Doxygen) and Python (Sphinx / Google style) **documentation** is
        updated accordingly.
    -   [ ] I have added or updated C++ and / or Python **unit tests** OR included **test
        results** (e.g. screenshots or numbers) here.
-   [ ] I will follow up and update the code if CI fails.
    
-   [ ] For fork PRs, I have selected **Allow edits from maintainers**.

### CODE REVIEW

File: cpp/open3d/t/geometry/TriangleMesh.cpp [hunk 1]
```diff
@@ -365,8 +365,8 @@ geometry::TriangleMesh TriangleMesh::FromLegacy(
                         .Reshape({-1, 3, 2}));
     }
 
-    // Convert material if legacy mesh only has one
-    if (mesh_legacy.materials_.size() == 1) {
+    // Convert material first material if legacy mesh has one
```
Comment (ssheorey) [inline cpp/open3d/t/geometry/TriangleMesh.cpp:368]: typo in comment

File: cpp/open3d/t/geometry/TriangleMesh.cpp [hunk 2]
```diff
@@ -393,10 +393,11 @@ geometry::TriangleMesh TriangleMesh::FromLegacy(
                     Image::FromLegacy(*mat.clearCoatRoughness));
         if (mat.anisotropy)
             tmat.SetAnisotropyMap(Image::FromLegacy(*mat.anisotropy));
-    } else if (mesh_legacy.materials_.size() > 1) {
+    }
+    if (mesh_legacy.materials_.size() > 1) {
         utility::LogWarning(
                 "Legacy mesh has more than 1 material which is not supported "
-                "by Tensor-based meshes.");
+                "by Tensor-based mesh. Only first material was converted.");
```
Comment (ssheorey) [inline cpp/open3d/t/geometry/TriangleMesh.cpp:400]: If available, print a description of material that was converted.

**Review by ssheorey — APPROVED** [review]
Thanks @errissa ! Looks good with minor nits.

File: cpp/open3d/t/geometry/TriangleMesh.cpp (hunk 1 above)
Comment (errissa) [reply cpp/open3d/t/geometry/TriangleMesh.cpp]: Done

File: cpp/open3d/t/geometry/TriangleMesh.cpp (hunk 2 above)
Comment (errissa) [reply cpp/open3d/t/geometry/TriangleMesh.cpp]: Done. Now outputting name of material converted.

**Pull request merged**